### ISSUE TITLE
Wrong paper found due to year filter in google schoar

### ISSUE THREAD

**Issue by SkqLiao** [justinribeiro/zotero-google-scholar-citation-count#32]
I've noticed that for newly published papers, the plugin is finding the wrong papers, and thus the wrong number of citations, due to google scholar's strange logic when adding year filtering.

Since the webpage opens when you enter the CAPTCHA, I noticed that the plugin would add a year range filter when searching for papers, but instead it found the wrong articles.

For example, for the paper "Machine learning in microscopy - insights, opportunities and challenges", if you don't add a year filter, the first paper found is correct, see [result](https://scholar.google.co.jp/scholar?q=Machine+learning+in+microscopy+%E2%80%93+insights,+opportunities+and+challenges&hl=zh-TW&as_sdt=0,5&as_vis=1) or the first image below. But if you add the time filter 2022-2024 (which is the behavior of the plugin), the wrong article is found, see [result](https://scholar.google.co.jp/scholar?q=Machine+learning+in+microscopy+%E2%80%93+insights%2C+opportunities+and+challenges&hl=zh-TW&as_sdt=0%2C5&as_vis=1&as_ylo=2022&as_yhi=2024) or the second image below.

My suggestion is whether it is possible to remove the year filter or add an extra step to determine if the title of the paper matches to avoid this situation happen. Thank you!

![image](https://github.com/user-attachments/assets/68e254d8-5a80-48ff-b830-b55fca19623c)

![image](https://github.com/user-attachments/assets/c4715cba-518c-49af-a161-b09c06648c38)

**Comment by justinribeiro** [comment]
Hummm, that is an interesting case. The year filter (which has been around for since I forked and rewrote this three years ago), clearly misses badly here. But the weirdness goes beyond that, as it doesn't hit with the additional information (the search string takes into account not only the year range, but also the authors):

https://github.com/justinribeiro/zotero-google-scholar-citation-count/blob/697052f465e2d1cbea9f62d4b666e0040d2344b2/src/chrome/content/gscc.js#L716-L718

```
https://scholar.google.com/scholar?hl=en&q=Machine%20learning%20in%20microscopy%20%E2%80%93%20insights,%20opportunities%20and%20challenges&as_epq=&as_occt=title&num=1&as_sauthors=Cunha+Latron+Bauer&as_ylo=2022&as_yhi=2026
```

Like, it misses by _a lot_, even with a good author string (which usually brings GS back inline).

I do have a better solution for this, but it's something I've hesitated to turn on (because it remove fuzzy matches, but given connector usage, maybe that's less of a problem).

Let's merge this other branch, we'll give it a spin.
